liquibase-cli/src/main/java/liquibase/integration/commandline/LiquibaseCommandLine.java

```diff
@@ -644,8 +644,9 @@ private void configureLogging(Level logLevel, File logFile) throws IOException {
 
         if (logFile != null) {
             if (fileHandler == null) {
-                fileHandler = new FileHandler(logFile.getAbsolutePath(), true);
-                fileHandler.setFormatter(new SimpleFormatter());
+                final LogFileFactory logFileFactory = Scope.getCurrentScope().getSingleton(LogFileFactory.class);
```

Comment:
Why do we need to be defining a new log-specific resource type? I get the limitation we are working around with s3, but it seems wrong to be singling out log files when this will be a general problem for other files and also how even **having** a log file is a not something that is assumed. There are lot of cases where there is no log file available.

Instead of introducing these new interfaces, could we update the ResourceAccessor/PathHandler/Resource APIs' openStream type methods to specify whether it should be an append or a replace operation?

Reply:
I added this log-specific resource type, because we want to append to log files by default. That is how Liquibase currently works. However, for S3, we're going to throw an exception if the log file the user is pointing to already exists in S3. So we need the ability to behave slightly differently for S3 than in all other circumstances.

> Instead of introducing these new interfaces, could we update the ResourceAccessor/PathHandler/Resource APIs' openStream type methods to specify whether it should be an append or a replace operation?

This would not allow us to have an S3 specific implementation for log file writing.

Reply:
An alternative option for resolving this would be to remove the log file plugin, and instead change the S3 code to throw an exception if the user tries to use the append open option and the file already exists. That would be a relatively transparent change as far as the open source code is concerned. What do you think of that?